src/routes/CreateTrip.jsx

```diff
@@ -1,75 +1,120 @@
-import { useState } from 'react'
-import { useNavigate } from 'react-router-dom';
-import { useUser } from '@clerk/clerk-react';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
 
+export default function CreateTripPage() {
+  const [tripDetails, setTripDetails] = useState({
+    photo: '',
+    name: '',
+    participants: '',
+    description: '',
+    date: '',
+  });
 
-export default function CreateTrip() {
-    const navigate = useNavigate();
-    const [title, setTitle] = useState("");
-    const [valid, setValid] = useState(false);
-    const { user } = useUser();
+  const [errors, setErrors] = useState({});
+  const [isSubmitted, setIsSubmitted] = useState(false);
 
-    let handleChange = (event) => {
-        setTitle(event.target.value)
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setTripDetails({ ...tripDetails, [name]: value });
+    setErrors({ ...errors, [name]: value.trim() === '' });
+  };
 
-        event.target.value.length >= 5 ? setValid(true) : setValid(false)
-    }
-
-    let createTrip = async () => {
-        if (!valid) {
-            return
-        }
-
-        const clerk_ID = user.id;
-
-        //get userID to do other api calls
-        const userID = await fetch(`${import.meta.env.VITE_API_URL}/getuserid/${clerk_ID}`)
-            .then(response => response.json());
+  const handleSubmit = () => {
+    const newErrors = {};
+    Object.keys(tripDetails).forEach((key) => {
+      if (!tripDetails[key].trim()) {
+        newErrors[key] = true;
+      }
+    });
 
-        const response = await fetch(`${import.meta.env.VITE_API_URL}/trips`, {
-            method: "POST", // Specify the HTTP method
-            headers: {
-                "Content-Type": "application/json",
-            },
-            body: JSON.stringify({
-                owner_id: userID.ID,
-                name: title, // Trip name
-                trip_desc: ""
+    setErrors(newErrors);
 
-            })
-        }).then(response => response.json()).then(response => response.trip_id);
+    if (Object.keys(newErrors).length === 0) {
+      // Сохраняем trip в localStorage
+      const storedTrips = JSON.parse(localStorage.getItem('trips') || '[]');
```

Comment:
instead of persisting to `localStorage`, persist to the database and fetch the trips when needed.